src/main/java/graphql/analysis/QueryTraverser.java

```diff
@@ -42,26 +44,26 @@ public class QueryTraverser {
     private final Collection<? extends Node> roots;
     private final GraphQLSchema schema;
     private final Map<String, FragmentDefinition> fragmentsByName;
-    private final Map<String, Object> variables;
+    private CoercedVariables coercedVariables;
 
     private final GraphQLCompositeType rootParentType;
 
     private QueryTraverser(GraphQLSchema schema,
                            Document document,
                            String operation,
                            Map<String, Object> variables) {
-        assertNotNull(document, () -> "document  can't be null");
+        assertNotNull(document, () -> "document can't be null");
         NodeUtil.GetOperationResult getOperationResult = NodeUtil.getOperation(document, operation);
         List<VariableDefinition> variableDefinitions = getOperationResult.operationDefinition.getVariableDefinitions();
         this.schema = assertNotNull(schema, () -> "schema can't be null");
         this.fragmentsByName = getOperationResult.fragmentsByName;
         this.roots = singletonList(getOperationResult.operationDefinition);
         this.rootParentType = getRootTypeFromOperation(getOperationResult.operationDefinition);
-        this.variables = coerceVariables(assertNotNull(variables, () -> "variables can't be null"), variableDefinitions);
+        this.coercedVariables = coerceVariables(assertNotNull(variables, () -> "variables can't be null"), variableDefinitions);
```

Comment:
For now, keep functionality similar to before

In the next PR (#2773), `QueryTraverser` will change to accept already coerced variables